test/src/820-https_s3/main

```diff
@@ -41,44 +38,44 @@ create_https_certificates() {
     # -nodes do not encrypt the private key
     # The key is $__/rootCA.key
     openssl req -x509 -new -nodes -key $SETTING_DIR/rootCA.key \
-	-sha256 -days 1024 -out $SETTING_DIR/rootCA.crt \
-	-subj "/C=CH/ST=CH/O=CERN/OU=CERN/CN=localhost" || return 2
+        -sha256 -days 1024 -out $SETTING_DIR/rootCA.crt \
+        -subj "/C=CH/ST=CH/O=CERN/OU=CERN/CN=localhost" || return 2
 
-    openssl genrsa -out $SETTING_DIR/localhost.key 2048 || return 3
+    openssl genrsa -out $SETTING_DIR/certs/private.key 2048 || return 3
 
     # generate a new certificate REQUEST
-    openssl req -new -sha256 -key $SETTING_DIR/localhost.key \
-	-subj "/C=CH/ST=CH/O=CERN/CN=localhost" \
-	-out $SETTING_DIR/localhost.csr || return 4
+    openssl req -new -sha256 -key $SETTING_DIR/certs/private.key \
+        -subj "/C=CH/ST=CH/O=CERN/CN=localhost" \
+        -out $SETTING_DIR/localhost.csr || return 4
 
     # expect a certificate request as input
     openssl x509 \
-	-req -in $SETTING_DIR/localhost.csr \
-	-CA $SETTING_DIR/rootCA.crt \
-	-CAkey $SETTING_DIR/rootCA.key \
-	-CAcreateserial \
-	-out $SETTING_DIR/certs/public.crt \
-	-days 500 -sha256 || return 5
-
-    cp $SETTING_DIR/localhost.key $SETTING_DIR/certs/private.key
+        -req -in $SETTING_DIR/localhost.csr \
+        -CA $SETTING_DIR/rootCA.crt \
+        -CAkey $SETTING_DIR/rootCA.key \
+        -CAcreateserial \
+        -out $SETTING_DIR/certs/public.crt \
+        -days 500 -sha256 || return 5
+
+    # we are making minio server use the new certificate
+    mkdir -p $SETTING_DIR/minio_server/certs
+    cp $SETTING_DIR/certs/* $SETTING_DIR/minio_server/certs/
 
     # we are making minio client use the new root CAs
     mkdir -p $SETTING_DIR/minio_client/certs/CAs
     cp $SETTING_DIR/rootCA.* $SETTING_DIR/minio_client/certs/CAs
 
-    sudo cp $SETTING_DIR/rootCA.key /etc/ssl/certs/cvmfs_https_test.key
-    sudo cp $SETTING_DIR/certs/public.crt /etc/ssl/certs/cvmfs_https_test1.crt
+    # we are making the whole system accept the new key
+    # useful for when we call curl in the bash script, libcurl does not care of this
+    sudo cp $SETTING_DIR/rootCA.crt /etc/ssl/certs/cvmfs_https_test.crt
```

Comment:
I think that works only on Ubuntu, on EL the certificate directory is rather /etc/pki/tls/certs